fusionbackend/src/main/java/io/fusion/fusionbackend/service/FieldInstanceService.java

```diff
@@ -0,0 +1,142 @@
+
+        if (!quantityDataType.equals(QuantityDataType.NUMERIC)) {
+            return existNoValues;
+        }
+
+        switch (fieldThresholdType) {
+            case MANDATORY:
+                return absoluteValuesCount == 2 && existPairwiseValues;
+            case OPTIONAL:
+                return (absoluteValuesCount == 2 && existPairwiseValues)
+                        || existNoValues;
+            case DISABLED:
+                return existNoValues;
+            default:
+                return false;
+        }
+    }
+
+    public void validate(FieldInstance fieldInstance) {
+        if (fieldInstance.getFieldSource() == null) {
+            throw new RuntimeException("Field instance must have a FieldSource");
+        }
+
+        final FieldThresholdType fieldThresholdType = fieldInstance.getFieldSource()
+                .getFieldTarget()
+                .getField()
+                .getThresholdType();
+
+        final Unit unit = unitService.getUnit(fieldInstance.getFieldSource().getSourceUnit().getId());
+        final QuantityDataType quantityDataType = unit.getQuantityType().getDataType();
+
+        if (!areThresholdsValid(fieldInstance, fieldThresholdType, quantityDataType)) {
+            LOG.warn("Thresholds of field instance with id {} are not valid.\r\n"
+                            + "Absolute Threshold: {}, ideal Threshold: {}, critical threshold: {}",
+                    fieldInstance.getId(), fieldInstance.getAbsoluteThreshold(),
+                    fieldInstance.getIdealThreshold(), fieldInstance.getCriticalThreshold());
+            throw new RuntimeException("Thresholds are not valid in every field instance");
+        }
+    }
+
+    private int countAbsoluteThresholdValues(FieldInstance fieldInstance) {
+        if (fieldInstance.getAbsoluteThreshold() == null) {
+            return 0;
+        }
+        return ((fieldInstance.getAbsoluteThreshold().getValueUpper() != null) ? 1 : 0)
+               + ((fieldInstance.getAbsoluteThreshold().getValueLower() != null) ? 1 : 0);
+    }
+
+    private int countIdealThresholdValues(FieldInstance fieldInstance) {
+        if (fieldInstance.getIdealThreshold() == null) {
+            return 0;
+        }
+        return ((fieldInstance.getIdealThreshold().getValueUpper() != null) ? 1 : 0)
+                + ((fieldInstance.getIdealThreshold().getValueLower() != null) ? 1 : 0);
+    }
+
+    private int countCriticalThresholdValues(FieldInstance fieldInstance) {
+        if (fieldInstance.getCriticalThreshold() == null) {
+            return 0;
+        }
+        return ((fieldInstance.getCriticalThreshold().getValueUpper() != null) ? 1 : 0)
+                + ((fieldInstance.getCriticalThreshold().getValueLower() != null) ? 1 : 0);
+    }
+
+    private boolean existPairwiseValues(final int absoluteValuesCount,
+                                        final int idealValuesCount,
+                                        final int criticalValuesCount) {
```

Comment:
here ist maby a shortcut with the XOR operator possible

!((a != null) ^ (b != null))